### PR TITLE
Specific interface for provider enhancement (1/3)

### PR DESCRIPTION
**Pull request by colerogers**
Change for the the specific app distro interface. Exposes `onNewTesterIosDevicePublished` from the `v2/alerts/appDistribution` namespace

Ex ~
```
import { onNewTesterIosDevicePublished } from "firebase-functions/v2/alerts/appDistribution";

// newTesterIosDevicePublished interface
export const func = onNewTesterIosDevicePublished((event) => { … });
export const func1 = onNewTesterIosDevicePublished("123456789", (event) => { … });
export const func2 = onNewTesterIosDevicePublished({ appId: "123456789" }, (event) => { … });
```

### CODE REVIEW

File: src/v2/providers/alerts/appDistribution.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    return handler(raw as AppDistributionEvent<NewTesterDevicePayload>);
+  };
+
+  func.run = handler;
+
+  // TypeScript doesn't recognize defineProperty as adding a property and complains
+  // that __endpoint doesn't exist. We can either cast to any and lose all type safety
+  // or we can just assign a meaningless value before calling defineProperty.
+  func.__trigger = 'silence the transpiler';
+  func.__endpoint = {} as ManifestEndpoint;
```
Comment (taeold) [inline src/v2/providers/alerts/appDistribution.ts:83]: nit* Both of these are unnecessary AFAIK.

File: src/v2/providers/alerts/appDistribution.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import { defineEndpoint, FirebaseAlertData } from '.';
```
Comment (taeold) [inline src/v2/providers/alerts/appDistribution.ts:1]: nit* I would've prefer having this point to a file instead, e.g. './alerts`. Saves me one jump to get to the actual source.

File: src/v2/providers/alerts/index.ts [hunk 3]
```diff
@@ -1 +1,5 @@
+import * as appDistribution from './appDistribution';
```
Comment (taeold) [inline src/v2/providers/alerts/index.ts:1]: nit* think this might be same as

```
export * as appDistribution from './appDistribution';
```

Also, why is this needed if we are explicitly declaring export path in package.json?

**Review by taeold — COMMENTED** [review]
mostly lgtm. waiting on some changes to flow from https://github.com/firebase/firebase-functions/pull/1020.

File: src/v2/providers/alerts/appDistribution.ts (hunk 1 above)
Comment (colerogers) [reply src/v2/providers/alerts/appDistribution.ts]: Removed

File: src/v2/providers/alerts/appDistribution.ts (hunk 2 above)
Comment (colerogers) [reply src/v2/providers/alerts/appDistribution.ts]: fixed

File: src/v2/providers/alerts/index.ts (hunk 3 above)
Comment (colerogers) [reply src/v2/providers/alerts/index.ts]: removed since we declare export path in the package file

File: src/v2/providers/alerts/appDistribution.ts [hunk 4]
```diff
@@ -1,10 +1,10 @@
-import { defineEndpoint, FirebaseAlertData } from '.';
-import { ManifestEndpoint } from '../../../common/manifest';
+import { getEndpointAnnotation, FirebaseAlertData } from './alerts';
 import { CloudEvent, CloudFunction } from '../../core';
 import * as options from '../../options';
 
-/** Data */
-/** Payload for a new tester device */
+/**
+ * The internal payload object that is wrapped inside a FirebaseAlertData object
```
Comment (taeold) [inline src/v2/providers/alerts/appDistribution.ts:6]: For consistency, I think we should end each comment with a period `.`

**Review by taeold — APPROVED** [review]
lgtm w/ a comment on adding punctuation on public doc strings.

File: src/v2/providers/alerts/appDistribution.ts (hunk 4 above)
Comment (colerogers) [reply src/v2/providers/alerts/appDistribution.ts]: Added periods to public comments

**Review by inlined — APPROVED** [review]
I love it
